### ISSUE TITLE
Workflow execution failure during dispatcher PR

### ISSUE THREAD

**Issue by burnout87** [oda-hub/dispatcher-plugin-nb2workflow#72]
During the development of https://github.com/oda-hub/dispatcher-app/pull/585 , the following workflow failed:

https://github.com/oda-hub/dispatcher-app/actions/runs/6149352313/job/16685039350

Please feel free to move the issue somewhere else more suitable if needed

**Comment by volodymyrss** [comment]
Actually I keep it open to confirm it's a duplicate

**Comment by dsavchenko** [comment]
This is not a duplicate, tests for both plugins were failing.
The complication is that it's not always reproducible.
And also it seems that the causes (or at least the manifestations in the logs) were probably different.

**Comment by burnout87** [comment]
I think the same error appeared again:

https://github.com/oda-hub/dispatcher-app/actions/runs/6484236816/job/17607602126

**Comment by dsavchenko** [comment]
Yes, it's the same error. And I can't reproduce it locally. I see lots of timeouts while polling dispather with oda-api in `test_full_stack`
The dispatcher is functional, though, and replies correctly, but after the request is timed out.

We discussed a bit that we want to profile and reduce latency if possible. But for the time being, as it only appears in pipeline, I propose https://github.com/oda-hub/dispatcher-plugin-nb2workflow/pull/73

**Comment by burnout87** [comment]
I re-run the workflow, and this time it completed

**Comment by dsavchenko** [comment]
Still, let's keep this issue open for some time

**Comment by burnout87** [comment]
I think I encountered again the same issue, is it related? A `TimeoutError` is mentioned.

https://github.com/oda-hub/dispatcher-app/actions/runs/7087471000/job/19287806505?pr=626

**Comment by burnout87** [comment]
ok, it did the same for me

**Comment by volodymyrss** [comment]
It would be better to have some better process starting/tracking behavior to avoid this. Though I suspect that this particular issue will not happen in production since if the service is not starting, the pod will be recreated.

**Comment by volodymyrss** [comment]
Still, let's keep for tracking open.

**Comment by dsavchenko** [comment]
> Though I suspect that this particular issue will not happen in production since if the service is not starting, the pod will be recreated.

Exactly, this mechanism is only used in tests

**Comment by burnout87** [comment]
looks like it happened again:

https://github.com/oda-hub/dispatcher-app/actions/runs/7130587275/job/19417305982

**Comment by volodymyrss** [comment]
> looks like it happened again:
> 
> https://github.com/oda-hub/dispatcher-app/actions/runs/7130587275/job/19417305982

production was down for some 15min, is something there calling it?
